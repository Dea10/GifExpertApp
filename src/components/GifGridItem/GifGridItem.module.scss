.card {
    box-shadow: 0 4px 8px 0 rgba(0,0,0,0.2);
    display: flex;
    flex-direction: column;
    justify-content: space-around;
    margin-top: 10px;
    width: 18%;

    .container {
        text-align: center;

        small {
            padding: 8px;
        }
    }
}