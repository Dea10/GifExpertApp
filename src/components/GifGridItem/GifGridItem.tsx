import React from 'react';
import styles from './GifGridItem.module.scss';

type GifGridItemProps = {
    title: string;
    url: string;
}

const GifGridItem = ({ title, url }: GifGridItemProps) => {

    return (
        <div className={ styles.card }>
            <img src={ url } alt={ title } />
            <div className={ styles.container }>
                <small>{ title }</small>
            </div>
        </div>
    );
}

export default GifGridItem;