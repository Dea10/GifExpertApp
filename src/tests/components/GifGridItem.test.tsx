import React from 'react';
import { shallow } from 'enzyme';
import GifGridItem from '../../components/GifGridItem/GifGridItem';

describe('GifGridItem tests', () => {
    const url = 'url';
    const title = 'title';
    
    const wrapper = shallow(<GifGridItem url={ url } title={ title } />);
    const img = wrapper.find('img');

    test('should render without errors', () => {
        expect(wrapper).toMatchSnapshot();
    });

    test('div should have card style class', () => {
        const div = wrapper.find('div').first()
        expect(div.prop('className')).toBe('card');
    });

    test('img src should render properly', () => {
        expect(img.prop('src')).toBe(url);
    });

    test('img alt should render properly', () => {
        expect(img.prop('alt')).toBe(title);
    });
});
