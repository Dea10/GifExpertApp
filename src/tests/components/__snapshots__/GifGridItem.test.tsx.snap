// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GifGridItem tests should render without errors 1`] = `
<div
  className="card"
>
  <img
    alt="title"
    src="url"
  />
  <div
    className="container"
  >
    <small>
      title
    </small>
  </div>
</div>
`;
